### ISSUE TITLE
Does not seems to work from the Gnomebar

### ISSUE THREAD

**Issue by Handl3vogn** [ifl0w/RandomWallpaperGnome3#66]
Cannot get it to work when pressing the screen and new wallpaper in my gnomebar.
And it seems to have problems auto fetching on schedule.

But request new wallpaper from the settings meny works.
Running Arch Linux using gnome 3.32.2 (running X11) and random wallpaper 2.4.1

**Comment by fikrimi23** [comment]
I also stumbled upon this issue for the last couple of days. Running on Arch 5.1.7-zen1-1-zen, Gnome shell 3.32.2 (specifically this 3.32.2+1+g24cb1c1aa-1: rt-scheduler, updated 6 days ago). 
- The auto-fetch didn't work alongside if I manually click "New Wallpaper" from gnomebar as stated above, no matter the source
- Request new wallpaper from inside settings still works.
- The file that downloaded seems to only fetch the header (?) as the name is present, but 0 bytes
- And this is from log 
```
Jun 10 20:00:31 ruby gnome-shell[1800]: Object Meta.Background (0x56040ddb7080), has been already deallocated — impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: == Stack trace for context 0x56040b09c310 ==
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #0   7fffb383aee0 b   resource:///org/gnome/shell/ui/background.js:702 (7f23b8008040 @ 22)
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #1   7fffb383c240 b   resource:///org/gnome/shell/ui/tweener.js:85 (7f23b83c88b0 @ 37)
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #2   7fffb383d5b0 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:208 (7f23b83d14c0 @ 54)
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #3   7fffb383d700 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:342 (7f23b83d1550 @ 1742)
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #4   7fffb383d7b0 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:355 (7f23b83d15e0 @ 100)
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #5   7fffb383d840 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:370 (7f23b83d1670 @ 10)
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #6   7fffb383d8c0 I   resource:///org/gnome/gjs/modules/signals.js:142 (7f23b83cb670 @ 386)
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #7   7fffb383d970 b   resource:///org/gnome/shell/ui/tweener.js:201 (7f23b83cb160 @ 159)
Jun 10 20:00:31 ruby org.gnome.Shell.desktop[1800]: #8   7fffb383da10 b   resource:///org/gnome/shell/ui/tweener.js:172 (7f23b83cb040 @ 15)
Jun 10 20:00:31 ruby gnome-shell[1800]: g_object_run_dispose: assertion 'G_IS_OBJECT (object)' failed
Jun 10 20:00:32 ruby gnome-shell[1800]: RWG3 [INFO] >> WallpaperController :: Requesting image: https://konachan.net/image/3eb8075f75447d4274baa4db266e023d/Konachan.com%20-%20218265%20brown_eyes%20brown_hair%20hat%20katou_megumi%20misaki_kurehito%20ribbons%20saenai_heroine_no_sodatekata%20scan%20short_hair.png
Jun 10 20:00:32 ruby gnome-shell[1800]: RWG3 [ERROR] >> WallpaperController :: Could not load new wallpaper.
```

**Comment by mrEDitor** [comment]
**TL;DR** It's Archlinux-specific, you can downgrade to [gnome-shell-1:3.32.2-2-x86_64](https://archive.archlinux.org/packages/g/gnome-shell/gnome-shell-1%3A3.32.2-2-x86_64.pkg.tar.xz) to fix this.

I am using archlinux too, and just logged error information from extension to log, got Gio.IOErrorEnum.NotSupported thrown at
https://github.com/ifl0w/RandomWallpaperGnome3/blob/635a9c6dc6199f4769b00237ff0867d2306dfdad/randomwallpaper%40iflow.space/wallpaperController.js#L132

Then I googled a little and come to [this ticket](https://bugs.archlinux.org/task/62860) and solution to downgrade gnome-shell.
Seems like the issue is not with extension but with latest Archlinux Gnome Shell build, especially in its GIO API.

**Comment by ifl0w** [comment]
Thanks for the research @mrEDitor. This seems to be the problem.

Sadly, I currently do not have the time to investigate the problem any further. However, I'll try to find a workaround if the issue remains with future updates of the gnome-shell. There might be a workaround by downloading the images without the GIO API but I would have to look into it further to confirm this.

**Comment by iXo** [comment]
I have the exact same problem. I could understand that this is something to a dependency (here gnome-shell), but if an extension is able to perform its job withing the settings window, but can't do it elsewhere in it code, to me it means that there could be a problem in the extension too. If the settings button is able to fetch the wallpaper, could this means that the code is different ?

**Comment by ifl0w** [comment]
@iXo That is a good thought, however, both, the settings and the extension use the same code for updating the wallpaper and only the execution environment is not the same. That is in line with the [ticket](https://bugs.archlinux.org/task/62860) mentioned by @mrEDitor. It seems to work in the GJS console and the preferences but not in the Gnome Shell.

Sadly, I do not have an idea of how to fix this problem, and I do not have the time to dig through the dependencies to search for the source of the issue.

**Comment by ifl0w** [comment]
You find a solution to this problem in this [ticket](https://bugs.archlinux.org/task/62860), and it worked for me on Arch Linux as well.
Unfortunately, I do not understand all side effects of this fix and do not take any responsibility for problems that might result from this fix.

Remove capabilities from gnome-shell with: 
`# setcap -r /usr/bin/gnome-shell`
Then reboot the system.

**Comment by iXo** [comment]
I can confirm that this manipulation on the gnome-shell executable works and make this extension works again. Thanks for the tip.
